.github/workflows/slither.yml

```diff
@@ -0,0 +1,45 @@
+name: Slither Analysis
+
+on:
+  pull_request:
+    branches: [main, release/**]
+  push:
+    branches: [main, release/**]
+    tags: [v*]
+
+permissions:
+  contents: read
+  security-events: write
+
+jobs:
+  analyze:
+    name: Slither Static Analysis
+    runs-on: smart-contracts-linux-large
```

Comment:
q: do we need large here? medium is not enough?

Reply:
Good point. Medium is enough, updating it.